testing/fpgad/test_fpgad_c.cpp

```diff
@@ -191,5 +205,113 @@ TEST_P(fpgad_fpgad_c_p, main_invalid) {
   EXPECT_NE(fpgad_main(2, argv), 0);
 }
 
+/**
+ * @test       main_symlink_gbs
+ * @brief      Test: fpgad_main
+ * @details    When fpgad_main is called with valid params<br>
+ *             but the input bitstream file is a symlink<br>
+ *             to a bitstream, fpgad returns non-zero.<br>
+ */
+TEST_P(fpgad_fpgad_c_p, main_symlink_gbs) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  strcpy(zero, "fpgad");
+  strcpy(one, "-n");
+  strcpy(two, "test_symlink");
+  strcpy(three, "-h");
+
+  char *argv[] = { zero, one, two, three };
+
+  ASSERT_EQ(symlink(tmp_gbs_, "test_symlink"), 0);
+  EXPECT_NE(fpgad_main(4, argv), 0);
+  EXPECT_EQ(unlink("test_symlink"), 0);
+}
+
+/**
+ * @test       main_symlink_log
+ * @brief      Test: fpgad_main
+ * @details    When fpgad_main is called with valid params<br>
+ *             but the output log file is a symlink<br>
+ *             to a file, fpgad returns non-zero.<br>
+ */
+TEST_P(fpgad_fpgad_c_p, main_symlink_log) {
```

Comment:
So for the log file and the pid file, both are now interpreted to be relative to the working directory; so main won't actually flag these as a failure.  The new fn resolve_dirs will catch .. and / and reject based on that, but otherwise the path is formed by concatenating the working dir plus the parameter given for -l.  I think we don't care whether the resulting path is a symlink, unless that symlink points to something **not** in the working directory.

Reply:
So the log and pid are set to be created in the current working directory. Ok, that's not what I thought your PR said. So to test this, it should create a symlink in the current working directory that is pointing to somewhere else, yes?

Reply:
Yes, that would be the correct check, which will currently fail.  @r-rojo you want to take this one while you're tweaking fpgad?